server/src/main/java/org/opensearch/index/remote/RemoteSegmentTransferTracker.java

```diff
@@ -675,5 +676,59 @@ public void writeTo(StreamOutput out) throws IOException {
             out.writeLong(bytesLag);
             out.writeOptionalWriteable(directoryFileTransferTrackerStats);
         }
+
+        @Override
+        public boolean equals(Object obj) {
```

Comment:
Are we going to compare 2 RemoteSegmentTransferTracker instances directly or indirectly anywhere?

Reply:
I see where you have used this. Ignore this comment.